docs/registry.md

```diff
@@ -0,0 +1,160 @@
+# Docker Registry
+
+This document describes the DOJO private Docker Registry, the token auth server, the CTFd verification endpoint, client usage, Image Puller, Production Checklist, troubleshooting, and future work.
+
+## Overview
+
+- Host: `registry.<domain>` served by `nginx-proxy`
+- Auth server: `/auth/token` on the same host (path-based) backed by `registry-auth`
+- Tokens: RS256 JWTs with a KID header that the registry uses to select the correct verification key
+- CTFd verify: `registry-auth` validates user creds and authorization by calling CTFd at `/pwncollege_api/v1/registry/verify`
+
+
+## Environment and Secrets
+
+- `REGISTRY_API_SECRET`: shared secret used by `registry-auth` to call CTFd’s verify endpoint
+  - Must be identical in both `ctfd` and `registry-auth`
+- `CTFD_URL`: URL of the CTFd service as seen from `registry-auth` (e.g., `http://ctfd:8000`)
+- Token parameters (consistent between services, single-host example):
+  - Registry: `REGISTRY_AUTH_TOKEN_REALM=https://registry.pwn.college/auth/token`
+  - Registry: `REGISTRY_AUTH_TOKEN_SERVICE=registry.pwn.college`
+  - Registry: `REGISTRY_AUTH_TOKEN_ISSUER=registry.pwn.college`
+  - Auth: `TOKEN_SERVICE=registry.pwn.college`, `TOKEN_ISSUER=registry.pwn.college`
+
+## Nginx Proxy Routes
+
+`dojo/nginx-proxy/etc/nginx/vhost.d/registry.localhost.pwn.college` illustrates path-based routing (replace with your domain in production):
```

Comment:
We should probably wrap all domain names used in the code in a variable that we can change in the `docker-compose.yml`. 

More generally, we need to easily configure the registry to work in a dev environment (without HTTPS with localhost addresses) and also in prod (with HTTPS with publicly resolvable addresses).